server/src/internalClusterTest/java/org/opensearch/indices/memory/breaker/ConcurrentSearchCircuitBreakerServiceIT.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.indices.memory.breaker;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.search.SearchService;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import static org.opensearch.test.OpenSearchIntegTestCase.Scope.TEST;
+
+/**
+ * Integration tests for InternalCircuitBreakerService with concurrent search enabled
+ */
+@OpenSearchIntegTestCase.ClusterScope(scope = TEST, numClientNodes = 0, maxNumDataNodes = 1)
+public class ConcurrentSearchCircuitBreakerServiceIT extends CircuitBreakerServiceIT {
```

Comment:
We need https://github.com/opensearch-project/OpenSearch/pull/9083 badly :-)

Reply:
yes exactly, but since that is not merged, i couldn't pull that in, I think we can merge that first, then we wouldnt need this extra class.

Reply:
Apologies if the recommendations are going in circles here but I think we still want to be able to run these tests with multiple slices too? The single slice case is just an edge case we want to test specifically.

Reply:
Thinking about this some more -- maybe it is best to just wait until #9083 and then randomize the slice count for this test?